core/src/main/scala/io/finch/Decode.scala

```diff
@@ -40,3 +41,36 @@ object Decode {
    */
   @inline def apply[A, CT <: String](implicit d: Aux[A, CT]): Aux[A, CT] = d
 }
+
+trait AdaptableDecode[A, CT] {
```

Comment:
Maybe we can move this under `Decode` companion object, by analogy with `ToResponse.Negotiable`?

Also, what do you think about renaming this to `Decode.Dispatchable`?

Reply:
Thinking out loud. I sort of like that we had `CT <: String` requirement on the `Decode` type-class. Perhaps, we can work it around and keep it in place for the sake of symmetry with `Encode`?

One thing we can do is to have a new type-class wrapping `Decode` (not returning it). Similar to how `ToResponse` and `Encode` relation is modeled.

Quick sketch. Let me know if you like it.

```scala
object Decode {
  
  trait Dispatchable[A, CT] {
    def apply(ct: Option[String], b: Buf, cs: Charset): Try[A]
  }
}
```

Reply:
Sounds good